src/listfunc.c

```diff
@@ -221,8 +221,7 @@ static Obj FuncAPPEND_LIST_INTR(Obj self, Obj list1, Obj list2)
     Int                 i;              /* loop variable                   */
 
     /* check the mutability of the first argument */
-    if (!IS_MUTABLE_OBJ(list1))
-        ErrorMayQuit("Append: <list1> must be a mutable list", 0, 0);
+    RequireMutable("Append", list1);
```

Comment:
As @wilfwilson pointed out, it's a bit sad that the conveyed information ("mutable *list*") is reduced. Perhaps if there was another optional argument to `RequireMutable`, like so:  `RequireMutable("Append", list1, "list");` to convey the "type"?

To implement that, I'd replace `Int showArgType` by `const char * alternateArgType`; if `alternateArgType` is NULL, then the usual `(not a %s)` is shown; otherwise `alternateArgType` is used instead (perhaps as another special case, if `alternateArgType` is an empty string, then also omit the space normally inserted before it).

Reply:
Thinking some more about it, of course just providing `"list"` there is not enough;  so I think instead, an invocation could like this: `RequireMutable("Append", list1, "a mutable list");`, and `showArgType` could be left as-is...

Reply:
I went with this kind of route in the end. For simplicity I assumed we will always start the message "a mutable ", but that could of course be changed later if ever needed. Sorry for the long new function, which ended up getting removed!